cloud/azure/azure_rm_virtualmachine.py

```diff
@@ -0,0 +1,1319 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2016 Matt Davis, <mdavis@ansible.com>
+#                    Chris Houseknecht, <house@redhat.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+DOCUMENTATION = '''
+---
+module: azure_rm_virtualmachine
+
+version_added: "2.1"
+
+short_description: Manage Azure virtual machines.
+
+description:
+    - Create, update, stop and start a virtual machine. Provide an existing storage account and network interface or
+      allow the module to create these for you. If you choose not to provide a network interface, the resource group
+      must contain a virtual network with at least one subnet.
+    - Currently requires an image found in the Azure Marketplace. Use azure_rm_virtualmachineimage_facts module
+      to discover the publisher, offer, sku and version of a particular image.
+
+options:
+    resource_group:
+        description:
+            - Name of the resource group containing the virtual machine.
+        required: true
+    name:
+        description:
+            - Name of the virtual machine.
+        required: true
+    state:
+        description:
```

Comment:
the states are confusing and overlap with other options, seems it would be better to just have present/absent and leave started=yes|no, allocated, restarted as diff options, or just merge them all in.